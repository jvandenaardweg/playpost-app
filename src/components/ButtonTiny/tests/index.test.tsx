import React from 'react';
import { fireEvent, render, RenderAPI } from 'react-native-testing-library';

import { ButtonTiny } from '../index';

describe('ButtonTiny', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;
    const onPressHandler = jest.fn();

    beforeAll(() => {
      wrapper = render(<ButtonTiny onPress={onPressHandler} label="Test" />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should fire onPress when the button is pressed', () => {
      fireEvent.press(wrapper.getByTestId('ButtonTiny-Button'));
      expect(onPressHandler).toHaveBeenCalledTimes(1);
    });
  });
});
