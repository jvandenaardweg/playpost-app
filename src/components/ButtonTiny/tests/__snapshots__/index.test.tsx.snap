// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTiny rendering should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 28,
        "height": 28,
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
  testID="ButtonTiny-Button"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 28,
          "flexDirection": "row",
          "height": 28,
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#040404",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 13,
            "lineHeight": 16,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;
